internal/buffer/save.go

```diff
@@ -24,6 +26,43 @@ import (
 // because hashing is too slow
 const LargeFileThreshold = 50000
 
+type saveRequest struct {
+	*Buffer
```

Comment:
It is really useful for it to be an anonymous field?

Reply:
At least it allowed direct access to the promoted functions from `*Buffer` like `sr.safeWrite(sr.path, sr.withSudo)`, without going over `sr.Buffer.safeWrite(sr.path, sr.withSudo)` or
```
b := sr.Buffer
b.safeWrite(sr.path, sr.withSudo)
```

Giving it the name e.g. `buffer` would require to do `sr.buffer.safeWrite(sr.path, sr.withSudo)` or
```
b := sr.buffer
b.safeWrite(sr.path, sr.withSudo)
```

In case it's more common to go with the second/explicit one, then I've no problem with that...just one line more.

Reply:
Why necessarily one line more? Just

```
sr.buf.safeWrite(sr.path, sr.withSudo)
```

Reply:
As already mentioned above, `sr.buffer.safeWrite(sr.path, sr.withSudo)` was also given as example and one-liner.
But I will use the shorter variable name, as you like. :)